### ISSUE TITLE
Add support for IHealthCheck for netcoreapp3.0 + net5.0

### ISSUE THREAD

**Issue by snakefoot** [NLog/NLog.Web#658]
Something similar to these:

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/tree/master/src

Maybe something like this:
- Setup static capture of NLog.InternalLogger-events on Warning and Error-Level.
- Keeps tracks of alert count per NLog Target within the last 10 minutes.
- Reports all NLog Targets (or NLog Core) that have generated alerts within the last 10 minutes.

**Comment by bakgerman** [comment]
is this still desired?  I am preparing pull requests for AspNetCore.Diagnostics.HealthChecks for IBM DB2 and MQ so I am somewhat familiar.  I can try and build a prototype.

**Comment by snakefoot** [comment]
I think it is still interesting to have diagnostics from the NLog-targets, and maybe also basic-statistics about logged-errors-per-second-by-application.

**Comment by bakgerman** [comment]
There are several locations to place the health check classes

One is in NLog.Web.AspNetCore itself, this will require 1 or 2 additional NuGet package dependencies added, for the IHealthCheck and related interfaces.  But as these are official Microsoft NuGet packages this may be OK.

One is in a separate project named NLog.Web.AspNetCore.HealthChecks.  This will have only the 1 or 2 NuGet dependencies, but will also have very few classes (an extension class, and a class to implement the actual check) so not sure it deserves a project.

The last one is to host the NLog.Web.AspNetCore.HealthChecks in the https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks repo, where there are many others...but then we will not be master maintainer of the project and cannot makes PR and release ourselves.  

As the HealthCheck NuGet packages state they support net std 2.0 and net 4.6.1, we 'may' be able to support that as well, but certainly core 3.1/5.0/6.0 they support.

**Comment by snakefoot** [comment]
Are the HealthChecks-dependencies not included by default when project is using:
```xml
<FrameworkReference Include="Microsoft.AspNetCore.App" />
```
I'm ok with only adding HealthCheck-support for only some platforms like NET6

**Comment by bakgerman** [comment]
Good point, yes if we skip netstd 20/4.6.1 then is auto included.  then we can leave in the project with no changes.
